ibft/controller/controller_sync.go

```diff
@@ -33,9 +34,18 @@ func (i *Controller) processSyncQueueMessages() {
 
 // ProcessSyncMessage processes sync messages
 func (i *Controller) ProcessSyncMessage(msg *network.SyncChanObj) {
+	i.logger.Info("ProcessSyncMessage: processing message")
 	var lastChangeRoundMsg *proto.SignedMessage
 	currentInstaceSeqNumber := int64(-1)
 	if i.currentInstance != nil {
+		if i.currentInstance.State() == nil {
+			i.logger.Warn("ProcessSyncMessage: i.currentInstance != nil but i.currentInstance.State() == nil")
+		} else if i.currentInstance.State().SeqNumber == nil {
+			i.logger.Warn("ProcessSyncMessage: i.currentInstance != nil && i.currentInstance.State() != nil " +
+				"but i.currentInstance.State().SeqNumber == nil")
+		}
+	}
+	if i.currentInstance != nil && i.currentInstance.State() != nil && i.currentInstance.State().SeqNumber != nil {
```

Comment:
what happens when if is false?

Reply:
@nivBlox If either of `i.currentInstance` or `i.currentInstance.State()` or `i.currentInstance.State().SeqNumber` is `nil`, we cannot do `i.currentInstance.State().SeqNumber.Get()`, so we don't update `lastChangeRoundMsg` and `currentInstanceSeqNumber`. Did I miss anything?